internal/cache/util/util_test.go

```diff
@@ -64,19 +64,32 @@ func (ut *utilTest) TearDown() {
 }
 
 func (ut *utilTest) assertFileAndDirCreation(file *os.File, err error) {
+	ut.assertFileAndDirStatsWithGivenFileAndDirPermissions(file, err, ut.fileSpec.Perm, 0700)
+}
+
+func (ut *utilTest) assertFileAndDirCreationWithGivenDirMode(file *os.File, err error, dirMode os.FileMode) {
```

Comment:
can we combine these functions (just keep 3rd one) and use that everywhere ?

Reply:
kept it deliberately like this so the tests using this function (eg: Test_CreateFile_ReadOnlyFile, Test_CreateFile_ReadWriteFile) which assume that directory is already created with default permissions dont need to have additional information that directory should have 700 permissions, its a good LLD practice to segregate responsibilities among different methods/classes as much as possible

Reply:
Just an fyi, generalising assertions into a common method is not a good practice, it breaks the AAA format where Assert should also be inside the test, also need for generalising assertions is a hint that we might be writing too many assertions in a single test which again is anti-pattern, ideally we should write one assertion per test but thats not feasible in real world scenarios but at least we can try to minimise and break the tests if needed
 if possible we should avoid these things in new tests
Short read on best practices : https://www.browserstack.com/guide/unit-testing-best-practices#:~:text=accomplish%20its%20purpose.-,Best%20Practices%20for%20Unit%20Testing,-These%20unit%20testing